pkg/clusteragent/admission/mutate/autoinstrumentation/config_test.go

```diff
@@ -164,6 +165,33 @@ func TestNewInstrumentationConfig(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:       "can provide DD_SERVICE from arbitrary label",
+			configPath: "testdata/filter_service_env_var_from.yaml",
+			expected: &InstrumentationConfig{
+				Enabled:            true,
+				EnabledNamespaces:  []string{},
+				DisabledNamespaces: []string{},
+				InjectorImageTag:   "0",
+				Version:            "v2",
+				LibVersions:        map[string]string{},
+				Targets: []Target{
+					{
+						Name: "name-services",
+						TracerConfigs: []TracerConfig{
+							{
+								Name: "DD_SERVICE",
+								ValueFrom: &corev1.EnvVarSource{
+									FieldRef: &corev1.ObjectFieldSelector{
+										FieldPath: "metadata.labels['app.kubernetes.io/name']",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
```

Comment:
It might be worth adding a test to reading the `DD_APM_INSTRUMENTATION_TARGTETS` env var to ensure the parsing works there too: https://github.com/DataDog/datadog-agent/blob/27fec2fa3aa613959ca4a9ce16daaefa69dadaba/pkg/clusteragent/admission/mutate/autoinstrumentation/config_test.go#L299

Reply:
Thanks for pointing this out.